chore: dependabot deps

mralj opened the pull request:
This PR updates most of the `yarn-project` dependabot alerted deps.

### On resolutions
I have tried to update all deps "properly", using `resolutions` as a fallback. This is when we depend on D, which has outdated D' for which Dependabot complains, and D doesn't have a newer version. 

An example of this is [docosaurus-plugin-ideal-image](https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-plugin-ideal-image/package.json) > sharp (outdated) > tar-fs (outdated) - meaning that `docusaurus-plugin-ideal-image` imports an outdated version of `sharp`, which in turn imports an outdated version of `tar-fs`, so we have to force this fix via `resolutions`

### Not updated deps

I could not update two dependencies because the author has not fixed the bug and published an update:
1. [bigint-buffer](https://github.com/no2chem/bigint-buffer/releases)
2. [private-ip](https://github.com/frenchbread/private-ip/releases) 

Both of these are transitive deps, further complicating the update.

spalladino reviewed (approved):
Looks good! 

On `bigint-buffer`, I see we're using it only via the chainsafe `discv5` and `enr` packages, but it seems they [removed this dependency earlier this year](https://github.com/ChainSafe/discv5/pull/306). Maybe we can update to those versions?

As for `private-ip`, it's only pulled in from `kad-dht`, which [removed it in early 2024](https://github.com/libp2p/js-libp2p/pull/2362). Perhaps we can update it as well? But more interestingly, it seems we're importing the package but not using it for anything, unless it has side effects? So maybe we could just remove it?

https://github.com/AztecProtocol/aztec-packages/blob/ce66e54a109c4d6eba35bd7ad10b9345c1cc69af/yarn-project/p2p/src/services/libp2p/libp2p_service.ts#L46

mralj commented:
> Looks good!
> 
> On `bigint-buffer`, I see we're using it only via the chainsafe `discv5` and `enr` packages, but it seems they [removed this dependency earlier this year](https://github.com/ChainSafe/discv5/pull/306). Maybe we can update to those versions?
>
> As for `private-ip`, it's only pulled in from `kad-dht`, which [removed it in early 2024](https://github.com/libp2p/js-libp2p/pull/2362). Perhaps we can update it as well? But more interestingly, it seems we're importing the package but not using it for anything, unless it has side effects? So maybe we could just remove it?
> 
> https://github.com/AztecProtocol/aztec-packages/blob/ce66e54a109c4d6eba35bd7ad10b9345c1cc69af/yarn-project/p2p/src/services/libp2p/libp2p_service.ts#L46

Good catch @spalladino, sorry for missing this 🤦

1. I have removed `kad-dht`
2. I will update `discv5` & `enr` in a separate PR, as the update yielded ~40 TS errors, so I think separate PR for this one is bettter :)

mralj committed:
chore: dependabot deps

This PR updates most of the `yarn-project` dependabot alerted deps.

### On resolutions
I have tried to update all deps "properly", using `resolutions` as a fallback. This is when we depend on D, which has outdated D' for which Dependabot complains, and D doesn't have a newer version.

An example of this is [docosaurus-plugin-ideal-image](https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-plugin-ideal-image/package.json) > sharp (outdated) > tar-fs (outdated) - meaning that `docusaurus-plugin-ideal-image` imports an outdated version of `sharp`, which in turn imports an outdated version of `tar-fs`, so we have to force this fix via `resolutions`

### Not updated deps

I could not update two dependencies because the author has not fixed the bug and published an update:
1. [bigint-buffer](https://github.com/no2chem/bigint-buffer/releases)
2. [private-ip](https://github.com/frenchbread/private-ip/releases)

Both of these are transitive deps, further complicating the update.

mralj commented:
@spalladino, updating `discv5`/`enr` seems to be a no-go at the moment. 
The issue is that both rely on `libp2p` types with breaking changes ([link](https://github.com/ChainSafe/discv5/blob/3c50f06b8a8ea4232dd2bc3ba60026bb14b2e132/packages/enr/package.json#L58)), which we import as well. 

Upgrading said packages (`libp2p/interface`/`libp2p/peer-id`) breaks a lot of code, and following this rabbit hole a bit it looks like it would imply upgrading the libp2p itself 🥲

spalladino commented:
Damn. WDYT about using [yarn patch](https://yarnpkg.com/cli/patch) to manually remove all bigint-buffer usage from the packages then? The other option is to just fork, since I doubt libp2p would accept a patch with that change for such an older version (though we could try).

mralj commented:
> Damn. WDYT about using [yarn patch](https://yarnpkg.com/cli/patch) to manually remove all bigint-buffer usage from the packages then? The other option is to just fork, since I doubt libp2p would accept a patch with that change for such an older version (though we could try).

I'll attack this tomorrow, atm. forking seems cleaner :)
